src/plugins/ui_feature/autofill.ts

```diff
@@ -222,13 +216,27 @@ export class AutofillPlugin extends UIPlugin {
           }
           const generator = this.createGenerator(xcs);
           for (let col = target.left; col <= target.right; col++) {
-            this.computeNewCell(generator, col, row, apply);
+            autofillCellsData.push(this.computeNewCell(generator, col, row));
           }
         }
         break;
     }
 
     if (apply) {
+      const borders: Record<UID, Border> = {};
+      const bordersPositions: Record<number, Zone[]> = {};
+      const cfNewRanges: Record<UID, string[]> = {};
+      const dvNewZones: Record<UID, Zone[]> = {};
+      for (const data of autofillCellsData) {
+        this.collectBordersData(data, borders, bordersPositions);
+        this.autofillMerge(data);
+        this.autofillCell(data);
+        this.collectConditionalFormatsData(data, cfNewRanges);
+        this.collectDataValidationsData(data, dvNewZones);
+      }
+      this.autofillBorders(borders, bordersPositions);
+      this.autofillConditionalFormats(cfNewRanges);
+      this.autofillDataValidations(dvNewZones);
```

Comment:
all these methods could receive the `activeSheetId`, thus calling  `getActiveSheetId` only once instead of once per cell per data type

Reply:
That's funny I preferred the original code, calls to `getActiveSheetId` are free and I'd rather not clutter method arguments if not needed. Either way that's a nitpick, I don't care that much 🤷